app/controllers/UserController.scala

```diff
@@ -28,22 +28,24 @@ class UserController @Inject()(val messagesApi: MessagesApi)
 
   val defaultAnnotationLimit = 1000
 
-  def current = SecuredAction { implicit request =>
-    Ok(Json.toJson(request.identity)(User.userPublicWrites(request.identity)))
+  def current = SecuredAction.async { implicit request =>
+    for {
```

Comment:
I think we have this pattern quite often, should we extract this at some point? (Not part of this PR, though.)

Reply:
```
implicit request =>
for {
  …
  js <- something.publicWrites(request.identity)
} yield Ok(userJs)
```

Reply:
see #2942